### DISCUSSION TITLE
Trace to logs - better query builder

### DISCUSSION THREAD

**Discussion by xificurC** [grafana/grafana#40379] [Observability]
Trace to logs jump allows specifying a set of tags that will be used as a Loki query. So IIUC a tag `app=myapp` on an opentelemetry span would create a Loki search `{app="myapp"}`. The default tags don't work for us, we don't have those in the application (maybe Grafana Agent is adding them? We don't use it). A more natural fit for off-the-shelf data would be the span's `service.name` field, which is easy to add as some Loki label. This has 2 problems though:

- it's not a tag. I'm not sure what it is, in JSON format it appears under `resource.attributes`
- the name is not a valid Loki label, as dots are not allowed

I wonder if I'm approaching this from a bad perspective or if this could be improved to help with more cases. E.g. the opentelemetry java agent doesn't allow adding custom common tags, but it allows specifying resource attributes.

**Comment by tonychoe** [comment]
Agreed, dots are breaking our trace-to-logs flow.

- Loki/Prometheus don't allow dots in the label name. See [data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels)
- However OpenTracing and OpenTelemetry attributes allow dots e.g. "host.name". See [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/semantic_conventions). 

Looks like this issue will address dots.
https://github.com/grafana/grafana/issues/31778

**Comment by aocenas** [comment]
I think this is basically the same as this feature request we already want to work on at some point https://github.com/grafana/grafana/issues/31778.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
